packages/block-editor/src/components/link-control/search-item.js

```diff
@@ -149,15 +149,27 @@ export const LinkControlSearchItem = ( {
 
 function getVisualTypeName( suggestion ) {
 	if ( suggestion.isFrontPage ) {
-		return 'front page';
+		return __( 'front page' );
 	}
 
 	if ( suggestion.isBlogHome ) {
-		return 'blog home';
+		return __( 'blog home' );
 	}
 
 	// Rename 'post_tag' to 'tag'. Ideally, the API would return the localised CPT or taxonomy label.
-	return suggestion.type === 'post_tag' ? 'tag' : suggestion.type;
+	const builtInLabels = {
+		post: __( 'post' ),
+		page: __( 'page' ),
+		post_tag: __( 'tag' ),
+		category: __( 'category' ),
+		attachment: __( 'attachment' ),
+	};
```

Comment:
Can you integrate with [this variable](https://github.com/WordPress/gutenberg/blob/ea0795048681be21302082c72144a37198b28fbe/packages/block-editor/src/components/link-control/search-item.js#L22-L28)? Something like this:

```javascript
const TYPES = {
	post: {
		icon: postList,
		label: __( 'Post' ),
	},
	page: {
		icon: pageIcon,
		label: __( 'Page' ),
	},
	post_tag: {
		icon: tag,
		label: __( 'Tag' ),
	},
	category: {
		icon: categoryIcon,
		label: __( 'Category' ),
	},
	attachment: {
		icon: file,
		label: __( 'Attachment' ),
	},
};
```